kernel/src/main/java/org/kframework/kast/KastFrontEnd.java

```diff
@@ -124,13 +135,59 @@ public int run() {
             }
             Module parsingMod = maybeMod.get();
 
-            K parsed = kread.prettyRead(parsingMod, sort, def, source, FileUtil.read(stringToParse));
+            if (options.genParser) {
+                try (ParseInModule parseInModule = RuleGrammarGenerator.getCombinedGrammar(parsingMod, true)) {
+                    try (Scanner scanner = parseInModule.getScanner()) {
+                        FileUtil files = this.files.get();
+                        File scannerFile = files.resolveTemp("scanner.l");
```

Comment:
Why is this in the temp directory?
Shouldn't this be inside the `-kompiled` directory?
Could you put all the related files inside a single directory? `-kompiled` already has many things in it, and it would make it easier to look for debugging info when files are a bit more organized.

Reply:
it's in the temp directory because they're temporary files; if you want to keep them around, you can pass --debug, but otherwise there's no need for them to be kept around after the program terminates.